purchase_open_qty/models/purchase_order.py

```diff
@@ -10,23 +10,38 @@
 class PurchaseOrderLine(models.Model):
     _inherit = 'purchase.order.line'
 
-    @api.depends('product_qty', 'qty_invoiced')
+    @api.depends('product_qty', 'qty_invoiced',
+                 'invoice_lines.invoice_id.state')
```

Comment:
In case of purchase_method = 'receive', it won't recompute.
I guess we have to add `qty_received` so the computation is triggered by every cases.
At the contrary, I am not sure the `invoice_lines.invoice_id.state` is useful?

Reply:
<code>qty_received</code> is another compute, so we need to add all the fields of the _compute_qty_received()'s depends.

Reply:
Are you sure? 
Here is an example on Odoo's code of a compute field depending on another one : 
https://github.com/odoo/odoo/blob/9.0/addons/account/models/account_move.py#L339

Reply:
just for have everything tied so anyone can follow the threads: this is fixed in https://github.com/OCA/purchase-workflow/pull/451